### PR TITLE
Auto Provision Identity Attributes From CA

### PR DESCRIPTION
**Pull request by andrewpmartinez**

### CODE REVIEW

File: controller/embedded/api-schema/entities/ca.put.json [hunk 1]
```diff
@@ -25,6 +25,10 @@
       "type": "boolean",
       "default": false
     },
+    "identityRoles": {
```
Comment (andrewpmartinez) [inline controller/embedded/api-schema/entities/ca.put.json]: @plorenz I believe this is a good attribute name but let me know if not.

File: tests/ca_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+// +build apitests
```
Comment (andrewpmartinez) [inline tests/ca_test.go]: @plorenz there are some testing facilities that you have built that I am not sure I am taking advantage of. Let me know if anything jumps out.

File: tests/entities.go [hunk 3]
```diff
@@ -557,3 +566,179 @@ func (entity *configValidatingService) validate(ctx *TestContext, c *gabs.Contai
...
+	tags                      map[string]interface{} `json:"tags"`
+
+	privateKey crypto.Signer     `json:"-"` //utility property, not used in API calls
+	publicCert *x509.Certificate `json:"-"` //utility property, not used in API calls
+}
+
+func newTestCa(identityRoles ...string) *ca {
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		panic(err)
```
Comment (andrewpmartinez) [inline tests/entities.go]: Not sure what to do w/ these types of errors. They *should* never happen.

File: controller/embedded/api-schema/entities/ca.put.json (hunk 1 above)
Comment (plorenz) [reply controller/embedded/api-schema/entities/ca.put.json]: Makes sense to me

File: tests/entities.go (hunk 3 above)
Comment (plorenz) [reply tests/entities.go]: I think for those kinds of errors, panic is fine. I would probably pass the test context in and use require.NoError, but only if it was convenient to do so

File: controller/embedded/api-schema/entities/ca.put.json (hunk 1 above)
Comment (ekoby) [reply controller/embedded/api-schema/entities/ca.put.json]: `defaultIdentityRoles` ?

**Pull request merged**